src/ir_Daikin.cpp

```diff
@@ -261,25 +268,457 @@ void IRDaikinESP::setSwingHorizontal(bool state) {
 bool IRDaikinESP::getSwingHorizontal() {
   return daikin[17] & 0x01;
 }
+
+
+void IRDaikinESP::setQuiet(bool state) {
+  if (state)
+    setBit(DAIKIN_BYTE_SILENT, DAIKIN_BIT_SILENT);
+  else
+    clearBit(DAIKIN_BYTE_SILENT, DAIKIN_BIT_SILENT);
+}
+
+bool IRDaikinESP::getQuiet() {
+  return (getBit(DAIKIN_BYTE_SILENT, DAIKIN_BIT_SILENT) > 0);
+}
+
+
+void IRDaikinESP::setPowerful(bool state) {
+  if (state)
+    setBit(DAIKIN_BYTE_POWERFUL, DAIKIN_BIT_POWERFUL);
+  else
+    clearBit(DAIKIN_BYTE_POWERFUL, DAIKIN_BIT_POWERFUL);
+}
+
+bool IRDaikinESP::getPowerful() {
+  return (getBit(DAIKIN_BYTE_POWERFUL, DAIKIN_BIT_POWERFUL) > 0);
+}
+
+
+void IRDaikinESP::setSensor(bool state) {
+  if (state)
+    setBit(DAIKIN_BYTE_SENSOR, DAIKIN_BIT_SENSOR);
+  else
+    clearBit(DAIKIN_BYTE_SENSOR, DAIKIN_BIT_SENSOR);
+}
+
+bool IRDaikinESP::getSensor() {
+  return (getBit(DAIKIN_BYTE_SENSOR, DAIKIN_BIT_SENSOR) > 0);
+}
+
+
+void IRDaikinESP::setEcono(bool state) {
+  if (state)
+    setBit(DAIKIN_BYTE_ECONO, DAIKIN_BIT_ECONO);
+  else
+    clearBit(DAIKIN_BYTE_ECONO, DAIKIN_BIT_ECONO);
+}
+
+bool IRDaikinESP::getEcono() {
+  return (getBit(DAIKIN_BYTE_ECONO, DAIKIN_BIT_ECONO) > 0);
+}
+
+
+void IRDaikinESP::setEye(bool state) {
+  if (state)
+    setBit(DAIKIN_BYTE_EYE, DAIKIN_BIT_EYE);
+  else
+    clearBit(DAIKIN_BYTE_EYE, DAIKIN_BIT_EYE);
+}
+
+bool IRDaikinESP::getEye() {
+  return (getBit(DAIKIN_BYTE_EYE, DAIKIN_BIT_EYE) > 0);
+}
+
+
+void IRDaikinESP::setMold(bool state) {
+  if (state)
+    setBit(DAIKIN_BYTE_MOLD, DAIKIN_BIT_MOLD);
+  else
+    clearBit(DAIKIN_BYTE_MOLD, DAIKIN_BIT_MOLD);
+}
+
+bool IRDaikinESP::getMold() {
+  return (getBit(DAIKIN_BYTE_MOLD, DAIKIN_BIT_MOLD) > 0);
+}
+
+
+void IRDaikinESP::setBit(uint8_t byte, uint8_t bitmask) {
+  daikin[byte] |= bitmask;
+  checksum();
+}
+
+
+void IRDaikinESP::clearBit(uint8_t byte, uint8_t bitmask) {
+  bitmask = ~bitmask;
+  daikin[byte] &= bitmask;
+  checksum();
```

Comment:
If we do `checksum()`s prior exporting the data (via returning a ptr, or sending the message) you don't need to do intermediate checksums, thus saving processing time.